modules/ServerAPI/src/server/routes/AuthorRoutes.coffee

```diff
@@ -964,4 +966,78 @@ exports.signupNewAuthorInternal = (author, cb) ->
 			cb json, null
 		else
 			cb null, json
+	)
+
+exports.updateAuthorRoles = (req, resp) ->
+	if req.session?.passport?.user?
+		user = req.session.passport.user
+	else
+		user =
+			username: 'anonymous'
+			roles: []
+	exports.updateAuthorRolesInternal req.body.newAuthorRoles, req.body.authorRolesToDelete, user, (err, response) ->
+		if err?
+			resp.statusCode = 500
+			resp.json err
+		else
+			resp.json response
+
+exports.updateAuthorRolesInternal = (newAuthorRoles, authorRolesToDelete, user, callback) ->
+	_ = require '../public/lib/underscore.js'
+	checkUserCanCreateOrEditAuthor user, (err, userCanCreate) ->
```

Comment:
This is a key fix. Previously this route was not checking permissions. Now it applies the same check as creating authors.

Reply:
Noob question: Where can I see what `checkUserCanCreateOrEditAuthor` is doing?

Reply:
`checkUserCanCreateOrEditAuthor` is a function defined in this same file which callsback to this function an error object (should be null) and a boolean as to whether the user making this request has the authority to edit author roles. It has a bunch of logic around whether the user is an admin and whether they have "editing roles" the logic is hard to read in the code but it's there.